drivers/wifi/eswifi/eswifi_core.c

```diff
@@ -175,27 +184,33 @@ static int eswifi_scan(struct eswifi_dev *eswifi)
 	err = eswifi_request(eswifi, cmd, strlen(cmd), eswifi->buf,
 			     sizeof(eswifi->buf));
 	if (err) {
+		eswifi->scan_cb(eswifi->iface, -EIO, NULL);
 		eswifi_unlock(eswifi);
-		return err;
+		return;
 	}
 
-	for (i = 0; i < sizeof(eswifi->buf); i++) {
-		if (eswifi->buf[i] == '#') {
+	data = eswifi_rsp_to_data(eswifi->buf, sizeof(eswifi->buf));
+	if (!data) {
+		eswifi->scan_cb(eswifi->iface, -EILSEQ, NULL);
```

Comment:
> -EILSEQ

That's an interesting error code here. I actually grepped the Zephyr codebase for it, and saw that it's used only in subsys/disk/disk_access_sdhc.c.

@rlubos: Do you remember our discussion that POSIX allegedly lacks a good error code for "invalid state [transition]"? This would be another candidate, in the sense of "illegal sequence [of state transitions]". Alas, that's not what it means: "Illegal byte sequence".

So, I have mixed feelings of it being used here. Here (just the same as in disk_access_sdhc.c) on low formal level it's correct: there was illegal byte sequence in the transport protocol (well, disk_access_sdhc.c uses it was invalid CRC). But why a client application (to which this error is fed) should care that there's a "transport protocol" which can go wrong? For it, it's rather a generic I/O error, and I'd recommend to use -EIO here.

Reply:
ok for -EIO